helix-rest/src/main/java/org/apache/helix/rest/server/resources/metadatastore/MetadataStoreDirectoryAccessor.java

```diff
@@ -165,8 +173,13 @@ public Response addShardingKey(@PathParam("realm") String realm,
   @Path("/metadata-store-realms/{realm}/sharding-keys/{sharding-key: .+}")
   public Response deleteShardingKey(@PathParam("realm") String realm,
       @PathParam("sharding-key") String shardingKey) {
+    if (shardingKey.charAt(0) != '/') {
+      shardingKey = "/" + shardingKey;
+    }
```

Comment:
Now this is duplicate code. Could you refactor this out to a private method? You could explain the JAX-RS behavior around slashes there.

Reply:
Done so.